cmd/kubelet/app/init_windows.go

```diff
@@ -34,26 +36,54 @@ const (
 // Ref: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/setpriority-method-in-class-win32-process
 func getPriorityValue(priorityClassName string) uint32 {
 	var priorityClassMap = map[string]uint32{
-		"IDLE_PRIORITY_CLASS":         uint32(64),
-		"BELOW_NORMAL_PRIORITY_CLASS": uint32(16384),
-		"NORMAL_PRIORITY_CLASS":       uint32(32),
-		"ABOVE_NORMAL_PRIORITY_CLASS": uint32(32768),
-		"HIGH_PRIORITY_CLASS":         uint32(128),
-		"REALTIME_PRIORITY_CLASS":     uint32(256),
+		"IDLE_PRIORITY_CLASS":         uint32(windows.IDLE_PRIORITY_CLASS),
+		"BELOW_NORMAL_PRIORITY_CLASS": uint32(windows.BELOW_NORMAL_PRIORITY_CLASS),
+		"NORMAL_PRIORITY_CLASS":       uint32(windows.NORMAL_PRIORITY_CLASS),
+		"ABOVE_NORMAL_PRIORITY_CLASS": uint32(windows.ABOVE_NORMAL_PRIORITY_CLASS),
+		"HIGH_PRIORITY_CLASS":         uint32(windows.HIGH_PRIORITY_CLASS),
+		"REALTIME_PRIORITY_CLASS":     uint32(windows.REALTIME_PRIORITY_CLASS),
 	}
 	return priorityClassMap[priorityClassName]
 }
 
+// addCurrentProcessToJobObjectAndSetPriorityClass creates a new Job Object
+// (https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects),
+// adds the current process to the job object, and specifies the priority
+// class for the job object to the specified value.
+// A job object is used here so that any spawned processes such as powershell or
+// wmic are created at the specified thread priority class.
```

Comment:
Examples? Didn't know kubelet spawned much 😮

Reply:
- https://github.com/kubernetes/kubernetes/blob/c1e69551be1a72f0f8db6778f20658199d3a686d/pkg/kubelet/winstats/perfcounter_nodestats.go#L245
- https://github.com/kubernetes/kubernetes/blob/542829d30d2e48abb802e4f4499dc63dd8076c3d/pkg/credentialprovider/gcp/metadata.go#L125
- https://github.com/kubernetes/kubernetes/blob/c1e69551be1a72f0f8db6778f20658199d3a686d/pkg/volume/azuredd/azure_common_windows.go#L37
- https://github.com/kubernetes/kubernetes/blob/dab6f6a43d5a41ef3e2aa11d4a85c45a268ac6cd/pkg/volume/util/util.go#L664
are a few

Reply:
Pretty much all volume mounting done is done with spawned powershell processes on Windows. This code is moving out of the kubelet to plugins that kubelet invokes too.

Reply:
Ahh, ok that makes sense.